src/rules/variableNameRule.ts

```diff
@@ -162,6 +176,21 @@ function walk(ctx: Lint.WalkContext<Options>): void {
         }
     }
 
+    function handleVariableDeclaration(node: ts.VariableDeclaration): void {
+        handleDeclaredVariable(node);
+
+        if (!ctx.options.allCapsForConst) {
+            return;
+        }
+
+        const declarationList = node.parent as ts.VariableDeclarationList;
+        const text = node.name.getText();
```

Comment:
this will lead to false negatives in destructuring
either restrict this check to identifiers or loop though all BindingElements